spec/controllers/comments_controller_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CommentsController, type: :controller do
+  let(:giveaway) { create(:giveaway) }
+  let(:comment) { create(:comment, giveaway: giveaway) }
+
+  describe 'GET new' do
+    it 'has a success status code' do
+      get :new, params: { giveaway_id: giveaway.id }
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'POST create' do
+    context 'with valid attributes' do
+      it 'creates new comment' do
+        expect do
+          post :create,
+               params: { giveaway_id: giveaway.id,
+                         comment: { commenter: 'Any Name', body: 'Sentence is here.' } }
+        end.to change(Comment, :count).by(1)
+      end
+    end
+  end
+
+  describe 'GET edit' do
+    it 'has a success status code' do
+      get :edit, params: { giveaway_id: giveaway.id, id: comment.id }
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'PATCH update' do
+    context 'with valid attributes' do
+      it 'edits the comment' do
+        patch :update, params: { giveaway_id: giveaway.id, id: comment.id,
+                                 comment: { commenter: comment.commenter, body: comment.body } }
+        expect(response).to have_http_status(:found)
+      end
+    end
+  end
+
+  describe 'DELETE destroy' do
+    it 'has a success status code' do
+      delete :destroy, params: { giveaway_id: giveaway.id, id: comment.id }
+      expect(response).to have_http_status(:found)
```

Comment:
I'd suggest using something like this: https://stackoverflow.com/a/19821867/11883546 to test the destroy action.